ui/src/app/job-list/job-list.component.ts

```diff
@@ -65,15 +66,23 @@ export class JobListComponent implements OnInit {
 
   reloadJobs(query: string) {
     if (this.streamSubscription) {
+      this.loading = true;
       this.streamSubscription.unsubscribe();
-      this.jobStream = new JobStream(this.jobManagerService,
+      const nextStream = new JobStream(this.jobManagerService,
           URLSearchParamsUtils.unpackURLSearchParams(query));
-      this.jobStream.loadAtLeast(initialBackendPageSize)
+      nextStream.loadAtLeast(initialBackendPageSize)
```

Comment:
Is there any way to cancel previous requests so that they don't block the next one?

Reply:
This is probably a larger change. It might involve switching our Promise usage over to Observables, as you unfortunately can't cancel a Promise.

Alternatively, we could return a cancellation callback (or accept that as a parameter); would need to plumb down to the job manager service.

Added a TODO.